homeassistant/components/comelit/light.py

```diff
@@ -0,0 +1,74 @@
+"""Support for lights."""
+from __future__ import annotations
+
+from typing import Any
+
+from aiocomelit import ComelitSerialBridgeObject
+from aiocomelit.const import LIGHT, LIGHT_OFF, LIGHT_ON
+
+from homeassistant.components.light import LightEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import ComelitSerialBridge
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Comelit lights."""
+
+    coordinator: ComelitSerialBridge = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities(
+        ComelitLightEntity(coordinator, device)
+        for device in coordinator.data[LIGHT].values()
+    )
+
+
+class ComelitLightEntity(CoordinatorEntity[ComelitSerialBridge], LightEntity):
+    """Light device."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    def __init__(
+        self, coordinator: ComelitSerialBridge, device: ComelitSerialBridgeObject
+    ) -> None:
+        """Init light entity."""
+        self._api = coordinator.api
+        self._device = device
+        super().__init__(coordinator)
+        self._attr_unique_id = f"{coordinator.api.host}-light-{device.index}"
```

Comment:
using a hostname or ip for the unique id is not allowed (_see [dev docs](https://developers.home-assistant.io/docs/entity_registry_index#unique-id-requirements)_)

Reply:
Unfortunately I can get the MAC only via web scrapping. No serial number anyware.
So I think the last resort is:

```python
        assert coordinator.config_entry
        self._attr_unique_id = (
            f"{coordinator.config_entry.unique_id}-light-{device.index}"
        )
```

Simone